lib/chatter_web/live/chat_live/index.ex

```diff
@@ -6,13 +6,29 @@ defmodule ChatterWeb.ChatLive.Index do
 
   @impl true
   def mount(_params, _session, socket) do
-    if connected?(socket), do: Message.subscribe()
-    {:ok, assign_default(socket)}
+    if connected?(socket), do: Message.subscribe_global()
+    socket =
+      socket
+      |> assign_default()
+      |> assign_messages()
+
+    subscribe_local_messages(socket)
+    {:ok, socket}
   end
 
   @impl true
-  def handle_info({Message, {:message, _action}, _message}, socket) do
-    {:noreply, assign_search_messages(socket)}
+  def handle_info({Message, {:message, :created}, message}, socket) do
+    {message, socket} |> created_response()
+  end
+
+  @impl true
+  def handle_info({Message, {:message, :updated}, message}, socket) do
+    {:noreply, stream_insert(socket, :messages, message, at: -1)}
+  end
+
+  @impl true
+  def handle_info({Message, {:message, :removed}, message}, socket) do
+    {:noreply, stream_delete(socket, :messages, message)}
```

Comment:
do we remove ?

Reply:
in this case handle_info used to responde for broadcast sygnal that message need to be removed from socket streams for this user